watch.go

```diff
@@ -71,7 +85,11 @@ func (fb *Firebase) Watch(notifications chan Event) error {
 		// if we're told to stop, close the response Body
 		go func() {
 			<-fb.stopWatching
+
+			fb.watchMtx.Lock()
 			closedManually = true
+			fb.watchMtx.Unlock()
```

Comment:
`closedManually` is not a property of `fb` and is scoped locally within this func. I'm not sure it's appropriate to use `fb.watchMtx` here. Perhaps `closedManually` itself could be replaced by something like this:

```go
type ConcurrentBool struct {
	b bool
	sync.RWMutex
}

func (b *ConcurrentBool) Value() bool {
	b.RLock()
	value := b.b
	b.RUnlock()
	return value
}

func (b *ConcurrentBool) Set(v bool) {
	b.Lock()
	b.b = v
	b.Unlock()
}
```

... or something using `sync/atomic`